docs/glossary.md

```diff
@@ -0,0 +1,65 @@
+| Consumer group           | Set of consumers that cooperate to consume data from some topics                                                                                                                             |
+| Key Value (KV) store     | Type of storage where data is stored by a specific key, allows for real-time lookups                                                                                                         |
+| TSDB                     | Time series database: part of V3IO                                                                                                                                                           |
+| V3IO                     | Iguazio real-time data layer, supports several formats including KV, Block, File, Streams, and more                                                                                                    |
+| V3IO shard               | Uniquely identified data sets within a V3IO stream. Similar to a Kafka partition                                                                                                              |
+| V3IO stream              | Streaming mechanism part of Iguazio's V3IO data layer. Similar to a Kafka stream                                                                                                              |
+
+## Standard ML Terms
+| Name                                       | Description          |   
+|--------------------------------------------------|---------------------------------------------------------------------------| 
+| Artifact                 | The input or output of experiments and model training runs that  can be used in many runs across the project.                                                                                  |
+| DAG                      | Directed acyclic graph                                                                                                                                                                        |
+| Feature engineering      | Apply domain knowledge and statistical techniques to raw data to extract more information out of data and improve performance of machine learning models                                      |
+| EDA                      | Exploratory data analysis. Used by data scientists to understand dataset via cleaning, visualization, and statistical tests                                                                   |
+| ML pipeline              | Pipeline of operations for machine learning. It can include loading data, feature engineering, feature selection, model training, hyperparameter tuning, model validation, and model deployment |
+| Feature                  | Raw characteristic or numerical value of a dataset                                                                                                                                            |
```

Comment:
Data field/vector definitions and metadata (name, type, stats, etc.), a dataset is a collection of features.

Reply:
updated